# SPHAK
Sequence-based Prediction of Host-spillover by Analysis of k-mers

sequence based approach

_Emerging pandemics are often driven by viral spillover events. However, predicting them remains difficult due to the complex interplay between viral genetics, host adaptability, and ecological dynamics. While traditional models rely on ecological or host trait data, recent advances using viral sequence data and machine learning have improved prediction accuracy. However, these approaches often lack a unified framework applicable across diverse viral families and host kingdoms, and are further constrained by the limited availability of comprehensive datasets. Here we introduce SPHAK, a simple proteome-based sequence similarity framework that could quantify spillover risk and predict viral family. By focusing on proteins, SPHAK is found to be a more effective way for identifying key protein patterns that can distinguish viral hosts. Since proteins are directly involved in infection, they could serve as a more informative probe than genome sequences. SPHAK is both accurate and generalizable, enabling its application across diverse viral families in plant and animal hosts. The predictive capacity of SPHAK supports its use in early warning systems and targeted surveillance, offering a practical tool to enhance pandemic preparedness and response._
